std-bits/table/src/main/java/org/enso/table/data/column/operation/StorageIterators.java

```diff
@@ -45,13 +46,17 @@ public static ColumnStorage<?> buildObjectOverDoubleStorage(
       boolean preserveNothing,
       Builder builder,
       DoubleBuildObjectOperation operation) {
-    long size = source.getSize();
     Context context = Context.getCurrent();
-    for (long index = 0; index < size; index++) {
-      if (preserveNothing && source.isNothing(index)) {
-        builder.appendNulls(1);
+    var iterator = source.iterator();
+    while (iterator.moveNext()) {
+      if (iterator.isNothing()) {
+        if (preserveNothing) {
+          builder.appendNulls(1);
+        } else {
+          operation.apply(builder, iterator.getIndex(), Double.NaN, true);
+        }
       } else {
-        operation.apply(builder, index, source.getItemAsDouble(index), source.isNothing(index));
+        operation.apply(builder, iterator.getIndex(), iterator.getItemAsDouble(), false);
       }
```

Comment:
This way of operating with index seems just nicer.

But again - if we are not using it perhaps it is not worth keeping it here - seems like too early 'future-proofing' (I know I'm guilty of this as well from time to time...) 

That is just an opinion though so feel free to ignore if you think it is better to have it already.